lib/ansible/modules/source_control/gitlab_deploy_key.py

```diff
@@ -94,140 +107,216 @@
+                'can_push': options['can_push']})
+
+        self.deployKeyObject = deployKey
+        if changed:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True, result="DeployKey should have been updated.")
+
+            try:
+                deployKey.save()
+            except Exception as e:
+                self._module.fail_json(msg="Failed to update a deployKey: %s " % e)
+            return True
+        else:
             return False
-    return True
+
+    '''
+    @param project Project Object
+    @param arguments Attributs of the deployKey
+    '''
+    def createDeployKey(self, project, arguments):
+        if self._module.check_mode:
+                self._module.exit_json(changed=True, result="DeployKey should have been created.")
+        try:
+            deployKey = project.keys.create(arguments)
+        except (gitlab.exceptions.GitlabCreateError) as e:
+            self._module.fail_json(msg="Failed to create a deployKey: %s " % to_native(e))
+
+        return deployKey
+
+    '''
+    @param deployKey Deploy Key Object
+    @param arguments Attributs of the deployKey
+    '''
+    def updateDeployKey(self, deployKey, arguments):
+        changed = False
+
+        for arg_key, arg_value in arguments.items():
+            if arguments[arg_key] is not None:
+                if getattr(deployKey, arg_key) != arguments[arg_key]:
+                    setattr(deployKey, arg_key, arguments[arg_key])
+                    changed = True
+
+        return (changed, deployKey)
+
+    '''
+    @param project Project object
+    @param key_title Title of the key
+    '''
+    def findDeployKey(self, project, key_title):
+        deployKeys = project.keys.list()
+        for deployKey in deployKeys:
+            if (deployKey.title == key_title):
+                return deployKey
+
+    '''
+    @param project Project object
+    @param key_title Title of the key
+    '''
+    def existsDeployKey(self, project, key_title):
+        # When project exists, object will be stored in self.projectObject.
+        deployKey = self.findDeployKey(project, key_title)
+        if deployKey:
+            self.deployKeyObject = deployKey
+            return True
+        return False
+
+    def deleteDeployKey(self):
+        if self._module.check_mode:
+                self._module.exit_json(changed=True, result="DeployKey should have been deleted.")
+
+        return self.deployKeyObject.delete()
 
 
 def main():
     module = AnsibleModule(
         argument_spec=dict(
-            api_url=dict(required=True),
-            access_token=dict(required=False, no_log=True),
-            private_token=dict(required=False, no_log=True),
+            server_url=dict(required=True, aliases=['api_url']),
+            verify_ssl=dict(required=False, default=True, type='bool', aliases=['enable_ssl_verification', 'validate_certs']),
```

Comment:
Same here, default parameter is **validate_certs**, the aliases should be removed.

We also simplify this by *not* adding `required=False`, and always starting with the type.
Since every parameter has a type, starting with this adds consistency.

Reply:
Will be fix in the next commit